contracts/general/src/lib.rs

```diff
@@ -41,6 +42,26 @@ pub struct Price {
     pub fraction_digits: Digits,
 }
 
+#[derive(Clone, Serialize, Deserialize)]
+#[serde(crate = "near_sdk::serde")]
+#[derive(Debug)]
+pub enum CampaignTypeArgs {
```

Comment:
@dmytro-shash why do we need to duplicate this struct?

Reply:
VestingArgs and CampaignTypeArgs allows us to pass arguments it ft_on_transfer more conveniently:

- not passing campaign_type (Supply or Borrow) via String, but using determined enum
- having Vesting as whole structure

Hence having more structured arguments while creating campaign